libraries/node-core-library/src/FileSystem.ts

```diff
@@ -232,14 +235,18 @@ export class FileSystem {
    * @param path - The absolute or relative path to the filesystem object.
    */
   public static getStatistics(path: string): fs.Stats {
-    return fsx.statSync(path);
+    return FileSystem._wrapException(() => {
+      return fsx.statSync(path);
+    });
   }
 
   /**
    * An async version of {@link FileSystem.getStatistics}.
    */
   public static getStatisticsAsync(path: string): Promise<fs.Stats> {
-    return fsx.stat(path);
+    return FileSystem._wrapExceptionAsync(() => {
+      return fsx.stat(path);
```

Comment:
Try throwing an exception on this line of code.  Does `getStatisticsAsync()` appear in the debugger's callstack?

If not, we can fix that by making the functions `async`, like this:
```ts
  public static async getStatisticsAsync(path: string): Promise<fs.Stats> {
    return await FileSystem._wrapExceptionAsync(() => {
      return fsx.stat(path);
    });
  }
```

Reply:
No. The error's stack comes from where the `Error` is constructed, not from where it's thrown.

For example:

```
> const { FileSystem } = require('./lib/index')
undefined
> FileSystem.readFile('foo')
Thrown:
{ Error: File does not exist: foo
    at Object.openSync (fs.js:443:3)
    at Object.readFileSync (fs.js:343:35)
    at FileSystem._wrapException (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:483:24)
    at Function._wrapException (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:708:20)
    at Function.readFileToBuffer (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:482:27)
    at FileSystem._wrapException (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:456:39)
    at Function._wrapException (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:708:20)
    at Function.readFile (E:\Code\rushstack1\libraries\node-core-library\lib\FileSystem.js:454:27) errno: -4058, syscall: 'open', code: 'ENOENT', path: 'foo' }
```

Reply:
I believe this is the exact problem that would get fixed by using `async`.